README.md

```diff
@@ -1,35 +1,116 @@
+
+1. Create the implementation of the [com.exactpro.sf.services.IService](https://github.com/exactpro/sailfish-core/blob/master/BackEnd/Core/sailfish-core/src/main/java/com/exactpro/sf/services/IService.java).
+The examples of implementing this interface can be found [here](https://github.com/exactpro/sailfish-core/tree/master/BackEnd/Service).
+If the protocol is already implemented in the Sailfish services you can simply use the dependencies on the service's artifact for that protocol.
+
+2. Create the **services.xml** configuration file that contains the description for services you can use from that "Connect" component.
+You can find the example [here](https://github.com/th2-net/th2-conn-generic/blob/master/conn-fix/src/main/plugin/cfg/services.xml).
+This file must contain:
+    + service name - the alias to use it from the "Connect" component;
+    + the full class name of the _com.exactpro.sf.services.IService_ interface implementation for the protocol;
+    + the full settings' class name for that protocol;
+    + the full validator's class name - the optional parameter. _Can be omitted_.
+
+3. Create a file with **VERSION** in the following format (parameters _plugin_alias_ and _name_ can be customized, but you will need to use a different folder on step 4):
+    ```yaml
+    lightweight: true
+    plugin_alias: th2_service
+    name: th2_service
+    build_number: 0
+    revision: 0
+    git_hash: 0
+    branch: fake
+    version: 3.2.0.0
+    core_version: 3.2.0
+    ```
+4. Create you own image based on the current one and put all files to the correct places in the base image:
+    + Create the following directory - **${workspace}/plugins/th2_service**.
+    _**${workspace}**_ - it is a folder from the "Connect" configuration.
+    If you use the _plugin_alias_ and _name_ different from _th2_service_ in the VERSION file correct the _th2_service_ folder name according to value you use.
+    Let's name that directory as **PLUGIN_DIRECTORY** for simplicity. This name will be used in future steps.
+    + Artifact with the service(s) implementation(s) and all its dependencies should be put to the following directory - **${PLUGIN_DIRECTORY}/libs**.
+    + The configuration file created on the step 2 should be put to the following directory - **${PLUGIN_DIRECTORY}/cfg**.
+    + The _VERSION_ file created on step 3 should be put to the following directory - **${PLUGIN_DIRECTORY}/**.
+
+## Pins
+
+Connect has 4 types of pins for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
+
+- incoming parsed messages // it will be removed in future
+- outgoing parsed messages // it will be removed in future
 - incoming raw messages
 - outgoing raw messages
 
-To send messages, a separate queue is used. The Connectivity component subscribes to that queue at the start and waits for the messages - to be sent to the target system - to come from the queue.
```

Comment:
**A separate queue is used to send messages.** The Connectivity component subscribes to that queue at the start and waits for the messages **(that will be sent to the target system)** to **be received** from the queue.